src/jsn.erl

```diff
@@ -920,6 +920,8 @@ key_get(_Key, _List, Default) ->
 %%------------------------------------------------------------------------------
 get_nth(_Index, [], Default) ->
     Default;
+get_nth(_Index, [{Tag, _}|_] = P, _Default) when Tag =/= struct ->
+    throw({error, {not_an_array, P}});
```

Comment:
I don't want this here, to be honest. If the `Path` is nonsense, I don't want a runtime exception. I want it to return the `Default`.